quarkus/runtime/src/main/java/org/keycloak/provider/quarkus/QuarkusClientConnectionFilter.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.provider.quarkus;
+
+import io.quarkus.vertx.web.RouteFilter;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.ext.web.RoutingContext;
+import org.keycloak.common.ClientConnection;
+import org.keycloak.common.util.Resteasy;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.services.filters.AbstractClientConnectionServletFilter;
+
+import javax.inject.Inject;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.ext.Provider;
+
+@PreMatching
+@Provider
+public class QuarkusClientConnectionFilter implements javax.ws.rs.container.ContainerRequestFilter {
+
+    @Inject
+    RoutingContext routingContext;
+
+    @Override
+    public void filter(ContainerRequestContext containerRequestContext) {
+        HttpServerRequest request = routingContext.request();
+        ClientConnection connection = createConnection(request);
+
+        AbstractClientConnectionServletFilter servletFilter = new AbstractClientConnectionServletFilter() {
```

Comment:
The fact we are creating instances of `AbstractClientConnectionServletFilter` on every request is not ideal. As well as duplicating code when pushing context data.

We could potentially override methods and have the main logic from the abstract class.

Reply:
That would be ideal if the same approach could be used in Wildfly, however it seems that this is not possible, servlet filter approach is not compatible with the vert.x filters, and it does not seem possible to use a common approach. So instantiating a _small_ instance seemed an acceptable alternative. Yet if you find a better alternative that would work for both, that would be indeed better.